src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -1,23 +1,79 @@
 package com.khutircraftubackend.product;
 
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
 @RestController
-@RequestMapping("/product")
+@RequestMapping("/products")
 @Slf4j
+@RequiredArgsConstructor
 public class ProductController {
+    private final ProductService productService;
 
+    @PostMapping("/create")
```

Comment:
no need in `/create`, just `@PostMapping` here

Reply:
Corrected.